launch/launch/frontend/parser.py

```diff
@@ -163,12 +215,10 @@ def load(
 
         try:
             filename = getattr(fileobj, 'name', '')
-            # file extension without leading '.'
-            extension = os.path.splitext(filename)[1][1:]
-
-            sorted_parsers = sorted(cls.frontend_parsers.items())
-            implementations = [v for k, v in sorted_parsers if k == extension] + [
-                v for k, v in sorted_parsers if k != extension
+            implementations = cls.get_parsers_from_filename(filename)
+            implementations += [
+                parser for _, parser in cls.frontend_parsers.items()
```

Comment:
@christophebedard nit: 
```suggestion
                parser for parser in cls.frontend_parsers.values()
```

Reply:
fixed in db9f1f2e9fce7b2268daa441bf13b0f9346206ac